packages/cli/src/commands/start.ts

```diff
@@ -340,15 +343,25 @@ export class Start extends BaseCommand {
 
 		if (config.getEnv('executions.mode') !== 'queue') return;
 
+		this.executionRecoveryService = Container.get(ExecutionRecoveryService);
+
+		this.executionRecoveryService.scheduleQueueRecovery();
```

Comment:
Do we have to init this immediately? In a multi main scenario, this would mean that upon startup, all main would immediately schedule the recovery right?

I think we should enable this only when `leader-takeover` happens, which is already present below.

Reply:
Also I don't think this is the best place to add this, since the method is called `initPruning` and we're also adding the execution recovery in it.

One thing I realized is that we have to:
1. Check if we're running single main. If so, init the processes here and let them run
2. If not multi main, then add the event listeners and let the `leader-takeover` dictate when to init the routines.

Reply:
Discussed sync on Friday - let's merge [this](https://github.com/n8n-io/n8n/pull/9761) and follow that pattern.

Reply:
See [6dba9c7](https://github.com/n8n-io/n8n/pull/9676/commits/6dba9c7d09983c26fd59ae28888b88ca5bea738c)